internal/utils.go

```diff
@@ -183,6 +191,26 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]map[string
 	return policyTemplates, nil
 }
 
+// checkPolicyTypeManifest determine if the manifest is policy-type manifest.
+func checkPolicyTypeManifest(manifest map[string]interface{}) bool {
+	apiVersion, foundAPIVersion, isStr := unstructured.NestedString(manifest, "apiVersion")
+	if !foundAPIVersion || isStr != nil {
+		return false
+	} else if strings.HasPrefix(apiVersion, "policy.open-cluster-management.io") {
+		return true // policy-type manifest contains policy api
+	}
+
+	return false
+}
+
+// buildFromPolicyTypeManifest take non-root policy-type manifest
+// and directly return as a policyTemplate.
+func buildFromPolicyTypeManifest(
```

Comment:
Since this code is only used once, I think you could get rid of this function and just use the code directly.